### PR TITLE
Update code for entities when same dim

### PR DESCRIPTION
**Pull request by chrisrichardson**
Fixes issue #158

### CODE REVIEW

File: cpp/dolfin/mesh/MeshEntity.h [hunk 1]
```diff
@@ -125,6 +125,9 @@ class MeshEntity
   /// dimension.
   inline std::size_t num_entities(std::size_t dim) const
   {
+    if (dim == _dim)
+      return 1;
+
```
Comment (garth-wells) [inline cpp/dolfin/mesh/MeshEntity.h]: could you add an `else` in?

File: cpp/dolfin/mesh/MeshEntity.h [hunk 2]
```diff
@@ -139,6 +142,9 @@ class MeshEntity
   ///         dimension.
   std::size_t num_global_entities(std::size_t dim) const
   {
+    if (dim == _dim)
+      return 1;
+
```
Comment (garth-wells) [inline cpp/dolfin/mesh/MeshEntity.h]: could you add an `else` in?

File: cpp/dolfin/mesh/MeshEntity.h [hunk 3]
```diff
@@ -152,6 +158,9 @@ class MeshEntity
   ///         The index for incident mesh entities of given dimension.
   const std::int32_t* entities(std::size_t dim) const
   {
+    if (dim == _dim)
+      return &_local_index;
+
```
Comment (garth-wells) [inline cpp/dolfin/mesh/MeshEntity.h]: could you add an `else` in?

File: cpp/dolfin/mesh/MeshEntity.h (hunk 1 above)
Comment (chrisrichardson) [reply cpp/dolfin/mesh/MeshEntity.h]: I could do - does it make it better? Some compilers will complain that "control reaches end of function returning void" or something...

**Pull request merged**